core/cloudflow-operator/src/main/scala/cloudflow/operator/action/Action.scala

```diff
@@ -43,12 +43,23 @@ sealed trait Action[+T <: ObjectResource] {
    * Returns the created or modified resource, or None if the resource is deleted.
    */
   def execute(client: KubernetesClient)(implicit ec: ExecutionContext, lc: LoggingContext): Future[Action[T]]
+
+  protected def recoverFromConflict[O](future: Future[O], client: KubernetesClient, f: KubernetesClient => Future[O])(
+      implicit ec: ExecutionContext
+  ): Future[O] =
+    future.recoverWith {
+      case e: K8SException if (e.status.code == Some(Action.ConflictCode)) => {
+        f(client)
```

Comment:
we might need to back off here, I think.

Reply:
it's not really necessary since a next get must get the new resource version, since we just stumbled on a conflict (we only use this for that purpose, but that's not really encoded, so in the general sense you are correct, but how we use it, it's fine).

Reply:
So what is implied is that `future` must always first get the resource.

Reply:
I've added a comment. Might be able to encode this, to do the get in this code to be safer. But this is just a internal helper so I'll leave it as is.